packages/language/src/linking/error.ts

```diff
@@ -11,15 +11,64 @@
 
 import {
   DiagnosticInfo,
+  Range,
   Severity,
   tokenToRange,
   tokenToUri,
 } from "../language-server/types";
 import { Token } from "../parser/tokens";
+import { Reference, SyntaxKind, SyntaxNode } from "../syntax-tree/ast";
 import { PLICodes } from "../validation/messages";
 import { PliValidationAcceptor } from "../validation/validator";
+import { CompilationUnit } from "../workspace/compilation-unit";
 import { QualifiedSyntaxNode } from "./qualified-syntax-node";
 
+function mergeRanges(range1: Range, range2: Range): Range {
+  return {
+    start: Math.min(range1.start, range2.start),
+    end: Math.max(range1.end, range2.end),
+  };
+}
```

Comment:
Suggestion: We can make this more general (by using an array), and we need to shield against `NaN`:
```suggestion
function mergeRanges(...ranges: Range[]): Range {
  return {
    start: Math.min(ranges.map(r => r.start).filter(n => !isNaN(n)),
    end: Math.max(ranges.map(r => r.end).filter(n => !isNaN(n)),
  };
}
```